sound/soc/intel/boards/sof_sdw.c

```diff
@@ -906,13 +912,18 @@ static int sof_card_dai_links_create(struct device *dev,
 		if (!name)
 			return -ENOMEM;
 
-		idisp_components[i].name = "ehdaudio0D2";
-		idisp_components[i].dai_name = devm_kasprintf(dev,
-							      GFP_KERNEL,
-							      "intel-hdmi-hifi%d",
-							      i + 1);
-		if (!idisp_components[i].dai_name)
-			return -ENOMEM;
+		if (ctx->idisp_codec) {
+			idisp_components[i].name = "ehdaudio0D2";
+			idisp_components[i].dai_name = devm_kasprintf(dev,
+								      GFP_KERNEL,
+								      "intel-hdmi-hifi%d",
+								      i + 1);
+			if (!idisp_components[i].dai_name)
+				return -ENOMEM;
+		} else {
+			idisp_components[i].name = "snd-soc-dummy";
+			idisp_components[i].dai_name = "snd-soc-dummy-dai";
+		}
 
 		cpu_name = devm_kasprintf(dev, GFP_KERNEL,
 					  "iDisp%d Pin", i + 1);
```

Comment:
@kv2019i   I tested it on TGL.  If HDMI_CODEC == NULL,  I will get error when doing playback with HDMI. you need to change cpu_name to another one instead of iDisp Pin.
````
[   46.911750] sof-audio-pci 0000:00:1f.3: ASoC: error at snd_soc_dai_hw_params on iDisp1 Pin: -22
[   46.911754]  iDisp1: ASoC: hw_params BE failed -22
[   46.911755]  HDMI1: ASoC: hw_params BE failed -22
````

Reply:
I mean ctx->idisp_codec == false

Reply:
@RanderWang That is expected failure -- there's no HDMI/DP codec attached, so error is expected. When ctx->idisp_codec is false, the HDMI controls will not be created and we have UCM rules to check this. So e.g. Pulseaudio will ignore the HDMI PCM nodes in this case. So normal applications should never hit this error. If we'd use a different cpu_name, then topology load would fail and this is specifically what we want to avoid.

Reply:
thanks,  I get it.